minikube/minikube.go

```diff
@@ -0,0 +1,117 @@
+package minikube
+
+import (
+	"os/exec"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	"os/user"
+	"k8s.io/client-go/rest"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	minikubeUsername = "minikube"
+	minikubeHost     = "192.168.99.100:8443"
+	minikubeCAFile   = "/.minikube/ca.crt"
+	minikubeKeyFile  = "/.minikube/apiserver.key"
+	minikubeCertFile = "/.minikube/apiserver.crt"
+)
+
+var (
+	errNotInstalled = errors.New("minikube is not installed. Please install minikube: https://github.com/kubernetes/minikube")
+	errNotRunning   = errors.New("minikube is not running. Run minikube using command 'minikube start'")
+	errPwFailed     = errors.New("Forwarding port of the minikube's docker registry failed")
+)
+
+// IsRunning is function to find out minikube is running or not
+func IsRunning() error {
+
+	_, err := exec.LookPath("minikube")
+	if err != nil {
+		return errNotInstalled
+	}
+
+	cmdCube := exec.Command("minikube", "ip")
+
+	err = cmdCube.Start()
```

Comment:
You can use `Run` instead of `Start`. This will automatically wait for the result